app/src/main/java/com/example/bookwormadventuresdeluxe2/LoginActivity.java

```diff
@@ -89,4 +145,157 @@ public void onClick(View view)
             }
         });
     }
+
+    /**
+     * Set Current User and attach Auth state listener on Start of Activity
+     */
+    @Override
+    protected void onStart()
+    {
+        currentUser = firebaseAuth.getCurrentUser();
+        firebaseAuth.addAuthStateListener(authStateListener);
+        super.onStart();
+    }
+
+    /**
+     * Detach Auth State Listener on idle Activity
+     */
+    @Override
+    protected void onPause()
+    {
+        if (firebaseAuth != null)
+        {
+            firebaseAuth.removeAuthStateListener(authStateListener);
+        }
+        super.onPause();
+    }
+
+    @Override
+    public void onClick(View view)
+    {
+        switch (view.getId())
```

Comment:
Can we add a default case to this? And add an exception if it is not expected to fallthrough?

Reply:
sure will do.